src/group_theory/free_product.lean

```diff
@@ -676,4 +677,146 @@ instance {ι : Type*} (G : ι → Type*) [∀ i, group (G i)] [hG : ∀ i, is_fr
     { intros g hfg, ext i x, simpa using hfg ⟨i, x⟩, }
   end, }
 
+/-- A free group is a free product of copies of the free_group over one generator. -/
+
+-- NB: One might expect this theorem to be phrased with ℤ, but ℤ is an additive group,
+-- and using `multiplicative ℤ` runs into diamond issues.
+@[simps]
+def _root_.free_group_equiv_free_product {ι : Type u_1} :
+  free_group ι ≃* free_product (λ (_ : ι), free_group unit) :=
+begin
+  refine monoid_hom.to_mul_equiv _ _ _ _,
+  exact free_group.lift (λ i, @free_product.of ι _ _ i (free_group.of unit.star)),
+  exact free_product.lift (λ i, free_group.lift (λ pstar, free_group.of i)),
+  { ext i, refl, },
+  { ext i a, cases a, refl, },
+end
+
+section ping_pong_lemma
+
+open_locale pointwise
+open_locale cardinal
```

Comment:
Can these be merged in 1 line. I think the syntax supports that.

Reply:
```suggestion
open_locale pointwise cardinal
```